x-pack/plugins/cloud_security_posture/public/common/hooks/use_cloud_posture_data_table/use_cloud_posture_data_table.ts

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { Dispatch, SetStateAction, useCallback } from 'react';
+import { type DataView } from '@kbn/data-views-plugin/common';
+import { BoolQuery, Filter } from '@kbn/es-query';
+import { CriteriaWithPagination } from '@elastic/eui';
+import { DataTableRecord } from '@kbn/discover-utils/types';
+import { useUrlQuery } from '../use_url_query';
+import { usePageSize } from '../use_page_size';
+import { getDefaultQuery, useBaseEsQuery, usePersistedQuery } from './utils';
+import { LOCAL_STORAGE_DATA_TABLE_COLUMNS_KEY } from '../../constants';
+import { FindingsBaseURLQuery } from '../../types';
+
+type URLQuery = FindingsBaseURLQuery & Record<string, any>;
+
+type SortOrder = [string, string];
+
+export interface CloudPostureDataTableResult {
+  setUrlQuery: (query: Record<string, any>) => void;
+  sort: SortOrder[];
+  filters: Filter[];
+  query: { bool: BoolQuery };
+  queryError?: Error;
+  pageIndex: number;
+  urlQuery: URLQuery;
+  setTableOptions: (options: CriteriaWithPagination<object>) => void;
+  handleUpdateQuery: (query: URLQuery) => void;
+  pageSize: number;
+  setPageSize: Dispatch<SetStateAction<number | undefined>>;
+  onChangeItemsPerPage: (newPageSize: number) => void;
+  onChangePage: (newPageIndex: number) => void;
+  onSort: (sort: string[][]) => void;
+  onResetFilters: () => void;
+  columnsLocalStorageKey: string;
+  getRowsFromPages: (data: Array<{ page: DataTableRecord[] }> | undefined) => DataTableRecord[];
+}
+
+/*
+  Hook for managing common table state and methods for the Cloud Posture DataTable
+*/
+export const useCloudPostureDataTable = ({
```

Comment:
This hook was created from the `use_cloud_posture_table` and it will replace it as we aim to remove old code in future PRs.